### PR TITLE
Rework section 4.3 methods

### PR DESCRIPTION
**Pull request by JohanBester**
Hi Folks, 

I have taken a stab at reworking section 4.3 on the 4 primary JavaScript methods. I have expanded the sections on .forEach(), .map(), and .filter().
I slightly reduced the section on .reduce().

Kindly please let me know what you think of these changes. Review the code and make your comments please.

Thanks!

Johan

### CODE REVIEW

File: Section04/4.3-ArrayMethods2.md [hunk 1]
```diff
@@ -49,20 +49,62 @@ These array methods **do not mutate** the array on which it is called (although
...
+The `forEach()` method executes a provided function once for each array element, iterating over the array items, in ascending order, without mutating the array. `forEach()` does not return anything. Code an implicit return statement to see output.
 
 ```javascript
 const nums = [3, 7, 11];
 const doubled = [];
 
 nums.forEach((num) => doubled.push(num * 2));
 
+// The above .forEach() statement has no return
+// To see any results sue console.log()
```
Comment (cbrantley4) [inline Section04/4.3-ArrayMethods2.md:61]: Spelling error: use

**Review by cbrantley4 — DISMISSED** [review]
There is one spelling mistake that I pointed out. I'm good with you merging this once that correction is made

**Review by matsinet — APPROVED** [review]
Ship it!

**Comment by JohanBester** [comment]
Ship it they said.

**Pull request merged**